src/formatDate.js

```diff
@@ -8,7 +8,84 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatObject = arrayInObject(fromFormat);
+  const toFormatObject = arrayInObject(toFormat);
+  const arrayOldDate = date.split(fromFormat[3]);
+  const dateObject = arrayDateFromObject(arrayOldDate, fromFormat);
+
+  if (fromFormatObject.yy !== toFormatObject.yy) {
+    if (fromFormatObject.yy.length === 4) {
+      dateObject.yy = dateObject.yy.slice(2);
+    } else {
+      if (+dateObject.yy < 30) {
+        dateObject.yy = '20' + dateObject.yy;
+      } else {
+        dateObject.yy = '19' + dateObject.yy;
+      }
+    }
+  }
```

Comment:
The condition `fromFormatObject.yy !== toFormatObject.yy` is always true because you are comparing the string values 'YYYY' or 'YY' to themselves. Instead, you should compare the lengths of the format strings to determine if a conversion is needed.